src/storage/src/hummock/compactor/context.rs

```diff
@@ -75,7 +78,55 @@ impl CompactorContext {
             memory_limiter: MemoryLimiter::unlimit(),
             task_progress_manager: Default::default(),
             await_tree_reg: None,
-            running_task_count: Arc::new(AtomicU32::new(0)),
+            running_task_parallelism: Arc::new(AtomicU32::new(0)),
+            max_task_parallelism: Arc::new(AtomicU32::new(u32::MAX)),
+        }
+    }
+
+    pub fn acquire_task_quota(&self, parallelism: u32) -> bool {
+        let running_u32 = self.running_task_parallelism.load(Ordering::SeqCst);
+        let max_u32 = self.max_task_parallelism.load(Ordering::SeqCst);
+
+        if parallelism + running_u32 > max_u32 {
+            return false;
+        }
+
+        if self
+            .running_task_parallelism
+            .compare_exchange(
+                running_u32,
+                running_u32 + parallelism,
+                Ordering::SeqCst,
+                Ordering::SeqCst,
+            )
+            .is_err()
+        {
+            return false;
+        }
```

Comment:
`acquire_task_quota` will return false not only when there is not enough quota but also when there is a competition. Is it better to change it to:
```
while parallelism + running_u32 < max_u32 {
        match self
        .running_task_parallelism
        .compare_exchange(
            running_u32,
            running_u32 + parallelism,
            Ordering::SeqCst,
            Ordering::SeqCst,
        ) {
            Ok(_) => return true;
            Err(old_running_u32) => running_u32 = old_running_u32;
        }
    }

    false
```

Reply:
Thanks a log, fix it